### PR TITLE
Delete Namespace: protect critical namespaces from deletion

### PR DESCRIPTION
**Pull request by alexshtin**
## What changed?

Protect critical namespaces from deletion.

## Why?

Safety net to prevent deletion of critical namespaces. Use:
```yaml
worker.protectedNamespaces:
  - value:
      - critical_namespace
      - just_very_important_namespace
```
dynamic config to list critical namespaces.

## How did you test it?

Added unit and functional tests.

## Potential risks

Change in `DeleteNamespace` workflow is not backward compatible (added new local activity), but this is a very short lived workflow (it starts actual deletion in abandon child mode) and shouldn't bring any real problems.

## Documentation

No.

## Is hotfix candidate?

No.

### CODE REVIEW

File: common/dynamicconfig/constants.go [hunk 1]
```diff
@@ -1015,6 +1015,11 @@ Default is 4.`,
 after all namespace resources (i.e. workflow executions) are deleted.
 Default is 0, means, namespace will be deleted immediately.`,
 	)
+	DeleteNamespaceBlockedNamespaces = NewGlobalTypedSetting(
```
[the remark this answers is not in the document]
Comment (alexshtin) [reply common/dynamicconfig/constants.go:1018]: I wish I can put a dot there `DeleteNamespace.BlockedNamespaces`. My concern was that w/o `DeleteNamespace` part it won't be clear what exactly this setting about. I like `ProtectedNamespaces` though.

File: service/worker/deletenamespace/workflow.go [hunk 2]
```diff
@@ -107,6 +107,42 @@ func validateParams(params *DeleteNamespaceWorkflowParams) error {
...
+
+	var la *localActivities
+
+	ctx1 := workflow.WithLocalActivityOptions(ctx, localActivityOptions)
+	var bns blockedNamespaces
+	err := workflow.ExecuteLocalActivity(ctx1, la.GetBlockedNamespacesActivity).Get(ctx, &bns)
+	if err != nil {
+		return fmt.Errorf("%w: GetBlockedNamespacesActivity: %v", errors.ErrUnableToExecuteActivity, err)
+	}
+	for _, blockedNsName := range bns.Names {
```
[the remark this answers is not in the document]
Comment (alexshtin) [reply service/worker/deletenamespace/workflow.go:137]: I would expect this list to be at max 5 elements. I believe that for up to 10 elements linear search is faster than anything else.

File: service/worker/deletenamespace/workflow.go [hunk 3]
```diff
@@ -135,24 +171,16 @@ func DeleteNamespaceWorkflow(ctx workflow.Context, params DeleteNamespaceWorkflo
...
 
-	// Disable namespace deletion if namespace is replicate because:
-	//  - If namespace is passive in the current cluster, then WF executions will keep coming from
-	//    the active cluster and namespace will never be deleted (ReclaimResourcesWorkflow will fail).
-	//  - If namespace is active in the current cluster, then it technically can be deleted (and
-	//    in this case it will be deleted from this cluster only because delete operation is not replicated),
-	//    but this is confusing for the users, as they might expect that namespace is deleted from all clusters.
-	if len(namespaceInfo.Clusters) > 1 {
-		return result, temporal.NewNonRetryableApplicationError(fmt.Sprintf("namespace %s is replicated in several clusters [%s]: remove all other cluster and retry", params.Namespace, strings.Join(namespaceInfo.Clusters, ",")), "", nil)
+	// Step 1.1. Validate namespace.
```
[the remark this answers is not in the document]
Comment (alexshtin) [reply service/worker/deletenamespace/workflow.go:179]: Step 2 is already there. And this is really like a sub step for 1.

File: service/worker/deletenamespace/workflow.go (hunk 2 above)
Comment (dnr) [inline service/worker/deletenamespace/workflow.go]: use `strings.Contains`?

Comment (alexshtin) [reply service/worker/deletenamespace/workflow.go]: `slices.Contains`, you mean. Yes, forgot about it.

**Pull request merged**